core/src/main/java/com/valtech/aem/saas/core/typeahead/DefaultTypeaheadService.java

```diff
@@ -0,0 +1,147 @@
+        GetQueryStringConstructor.builder()
+            .query(new TypeaheadTextQuery(typeaheadPayload.getText()))
+            .query(new DefaultLanguageQuery(typeaheadPayload.getLanguage()));
+    if (MapUtils.isNotEmpty(typeaheadPayload.getFilterEntries())) {
+      FiltersQuery.FiltersQueryBuilder filtersQueryBuilder = FiltersQuery.builder();
+      typeaheadPayload.getFilterEntries().forEach(filtersQueryBuilder::filterEntry);
+      builder.query(filtersQueryBuilder.build());
+    }
+    return builder.build().getQueryString();
+  }
+
+
+  private void validateFilterFields(@NonNull TypeaheadPayload typeaheadPayload) {
+    if (ArrayUtils.isNotEmpty(configuration.typeaheadService_allowedFilterFields())) {
+      List<String> forbiddenFilterFields = typeaheadPayload.getFilterEntries().keySet().stream()
+          .filter(field -> !ArrayUtils.contains(configuration.typeaheadService_allowedFilterFields(), field))
+          .collect(Collectors.toList());
+      if (CollectionUtils.isNotEmpty(forbiddenFilterFields)) {
+        throw new IllegalArgumentException(
+            String.format("The following filter field names are not allowed: %s", forbiddenFilterFields));
+      }
+    }
+  }
+
+  @Override
+  public List<String> getAllowedFilterFields() {
+    return Optional.ofNullable(configuration.typeaheadService_allowedFilterFields())
+        .map(Arrays::stream)
+        .orElse(Stream.empty())
+        .collect(Collectors.toList());
+  }
+
+  private String getApiUrl(String index) {
+    return String.format("%s%s/%s%s",
+        searchServiceConnectionConfigurationService.getBaseUrl(),
+        configuration.typeaheadService_apiVersionPath(),
+        index,
+        configuration.typeaheadService_apiAction());
+  }
+
+  @Activate
+  @Modified
+  private void activate(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  @ObjectClassDefinition(name = "Search as a Service - Typeahead Service Configuration",
+      description = "Typeahead Api specific details.")
+  public @interface Configuration {
+
+    String DEFAULT_API_ACTION = "/typeahead";
+    String DEFAULT_API_VERSION_PATH = "/api/v3";
+
+    String FF_LANGUAGE = "language";
+    String FF_DOMAIN = "domain";
+    String FF_REPOSITORY_PATH_URL = "repository_path_url";
+
+    @AttributeDefinition(name = "Api version path",
+        description = "Path designating the api version",
+        type = AttributeType.STRING)
+    String typeaheadService_apiVersionPath() default DEFAULT_API_VERSION_PATH;
+
+    @AttributeDefinition(name = "Api action",
+        description = "Path designating the action",
+        type = AttributeType.STRING)
+    String typeaheadService_apiAction() default DEFAULT_API_ACTION;
+
+    @AttributeDefinition(name = "Allowed filter fields",
+        description = "List of field names that can be used in filter queries",
+        type = AttributeType.STRING)
+    String[] typeaheadService_allowedFilterFields() default {
```

Comment:
I think we discussed this once: Is it required limit those filter really? Worst case search will respond with an error. Reason I am asking: Fields can be added in runtime on search admin. 
We don't want to have a new aem release than as ogi configs are immutable so we would need through cloud manager